File: backend/src/main/java/com/example/estimateserver/model/WorkItem.java
package com.example.estimateserver.model;

import jakarta.persistence.*;
import java.util.UUID;

@Entity
@Table(name = "work_items")
public class WorkItem {
    @Id
    private String id = UUID.randomUUID().toString();

    @Column(nullable = false)
    private String projectId;

    @Column(nullable = false)
    private String name;

    private Integer stage = 1;
    private Double laborHours = 0.0;
    private Double hourlyRate = 0.0;
    private Double materialCost = 0.0;
    private Boolean isCompleted = false;
    private String notes = "";

    @Version
    private Long version;

    @Column(nullable = false)
    private String userId;

    public WorkItem() {}

    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getProjectId() { return projectId; }
    public void setProjectId(String projectId) { this.projectId = projectId; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public Integer getStage() { return stage; }
    public void setStage(Integer stage) { this.stage = stage; }

    public Double getLaborHours() { return laborHours; }
    public void setLaborHours(Double laborHours) { this.laborHours = laborHours; }

    public Double getHourlyRate() { return hourlyRate; }
    public void setHourlyRate(Double hourlyRate) { this.hourlyRate = hourlyRate; }

    public Double getMaterialCost() { return materialCost; }
    public void setMaterialCost(Double materialCost) { this.materialCost = materialCost; }

    public Boolean getIsCompleted() { return isCompleted; }
    public void setIsCompleted(Boolean isCompleted) { this.isCompleted = isCompleted; }

    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes; }

    public Long getVersion() { return version; }
    public void setVersion(Long version) { this.version = version; }

    public Double getLaborCost() {
        return laborHours * hourlyRate;
    }

    public Double getTotalCost() {
        return getLaborCost() + materialCost;
    }
}

File: app/src/main/java/com/example/myapplication/data/models/WorkItem.kt
package com.example.myapplication.data.models

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import java.util.Date
import java.util.UUID

@Entity(
    tableName = "work_items",
    indices = [Index(value = ["projectId"])],
    foreignKeys = [
        ForeignKey(
            entity = Project::class,
            parentColumns = ["id"],
            childColumns = ["projectId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class WorkItem(
    @PrimaryKey
    val id: String = UUID.randomUUID().toString(),
    val projectId: String,
    val name: String,
    val stage: Int = 1,
    val laborHours: Double = 0.0,
    val hourlyRate: Double = 0.0,
    val materialCost: Double = 0.0,
    val isCompleted: Boolean = false,
    val completedAt: Date? = null,
    val notes: String = ""
) {
    // Стоимость работы (часы × ставка)
    val laborCost: Double get() = laborHours * hourlyRate

    // Общая стоимость (работа + материалы)
    val totalCost: Double get() = laborCost + materialCost
}